src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -246,86 +251,75 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
       return;
     }
 
-    this._loaded = true;
-
-    await this.updateComplete;
+    this._large = this._medium = this._small = false;
+    this._radius = this.clientWidth / 3.9;
 
-    let radius = this.clientWidth / 3.2;
-    this._broadCard = this.clientWidth > 390;
-
-    if (radius === 0) {
-      radius = 100;
+    if (this.clientWidth > 450) {
+      this._large = true;
+    } else if (this.clientWidth < 350) {
+      this._small = true;
+    } else {
+      this._medium = true;
     }
 
-    (this.shadowRoot!.querySelector(
-      "#thermostat"
-    ) as HTMLElement)!.style.height = radius * 2 + "px";
-
-    const loaded = await loadRoundslider();
-
-    this._roundSliderStyle = loaded.roundSliderStyle;
-    this._jQuery = loaded.jQuery;
+    await this.updateComplete;
```

Comment:
What triggers the update we're awaiting?

Reply:
https://lit-element.polymer-project.org/guide/lifecycle#updatecomplete

Reply:
Oh, sorry!
Github was folding the source, so I thought this was inside `firstUpdated`.